### PR TITLE
Project draft

### PR DESCRIPTION
**Pull request by danielamoreno699**
**Description of the Project:** 
The project is a single-page application (SPA) built with React and Redux.
The data retrieved from the API is  stored in the Redux store.
The  data provided by the API is  implemented by using a   filter stateless component 
Every page (the main page and the pages for each item) has an unique route within the SPA.

**In this milestone project, I:**

- Selected a stock market API  which allows to get numeric values based on a parameter. 
- Folowed the designed provided by the figma in terms of typography, font and layout and included the credit in the README.md
- Created  2 pages. the Home page and the Detail Page. In the Home page, the list of categories is rendered. They can be filtered by  the symbol of the stock. Along with the category name, some numeric values are displayed . When the user clicks (or taps) on a category item, the application navigates to the details page.

-In the details page, the webapp retrieves data from the API to show detailed data in the category.
When the user clicks on the "Back" button (<), the user navigates to the home page.
-Applied Testing requirements. Created Mocks to access API and also thta connects to Redux

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,164 @@
...
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you're on your own.
+## 🚀 Live demo <a name="live-demo"></a>
 
-You don't have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn't feel obligated to use this feature. However we understand that this tool wouldn't be useful if you couldn't customize it when you are ready for it.
+Under construction.
+<!-- - [Live Demo Link](https://j-c-s-v.github.io/webpack-to-do-list/) -->
 
-## Learn More
+loom Video : https://www.loom.com/share/2600f52416f84137ab1873b77db989db
```
Comment (julie-ify) [inline README.md]: - [ ] Kindly deploy your app and provide the link here.

File: src/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    if (status === 'idle' && data.length === 0) {
+      dispatch(fetchData());
+    }
+  }, [status, dispatch, data]);
+
+  const onHandleSelect = (ticker) => {
+    dispatch(selectStock(ticker));
+    navigate(`/Details/${ticker}`);
+  };
+  return (
```
Comment (julie-ify) [inline src/components/Home.jsx]: - [ ] Kindly implement the search functionality as stated in the requirements.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @danielamoreno699,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/stockSlice.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  initialState,
+  reducers: {
+    selectStock: (state, action) => {
+      const selectedStock = action.payload;
+      return {
+        ...state,
+        stockInfo: selectedStock,
+      };
+    },
+  },
```
Comment (GzimAsani) [inline src/redux/stockSlice.js:46]: - [ ] I still don't on the webpage a way where I can search/filter what I want to see, kindly implement this requirement
image
image

File: README.md [hunk 4]
```diff
@@ -1,70 +1,166 @@
...
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you're on your own.
+## 🚀 Live demo <a name="live-demo"></a>
 
-You don't have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn't feel obligated to use this feature. However we understand that this tool wouldn't be useful if you couldn't customize it when you are ready for it.
 
-## Learn More
+ [Live Demo Link](https://j-c-s-v.github.io/webpack-to-do-list/)
```
Comment (GzimAsani) [inline README.md]: - [ ] Kindly add here the link of your current application. This link is for the todo app project

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @danielamoreno699                    :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with adding the live demo:+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ OPTIONAL ] I know that git-guardian linter is not part of the project, but it is advisable to make sure no important data is being leaked



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 4 above)
Comment (bobb-Rob) [inline README.md]: - [ ] You still have the wrong link here. The attached link is for the to-do app project. Please update this demo section with a link for this project. I see a [new link](https://elegant-treacle-bdd353.netlify.app/Home) which I believe is for this project ut the link leads to a broken 404 page.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️

Hi @danielamoreno699, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @danielamoreno699 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**